include/zephyr/sys/uuid.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UUID_H_
+#define ZEPHYR_INCLUDE_SYS_UUID_H_
+
+/**
+ * @file
+ *
+ * @brief Utility functions for the generation and parsing of Universal Unique Identifier.
+ * @details This driver is compliant with RFC9562: https://datatracker.ietf.org/doc/rfc9562/
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup uuid UUID
+ * @ingroup utilities
+ * @{
```

Comment:
Suggest to add `@since` and `@version 0.1.0` to indicate an experimental API, so that we can modify it more easily in the coming months in case we need to modify it. 

See https://docs.zephyrproject.org/latest/develop/api/overview.html#api-overview for more info